### PR TITLE
[batch] Restore read-only cloudfuse

### PR DESCRIPTION
**Pull request by jigold**

### CODE REVIEW

File: batch/batch/front_end/front_end.py [hunk 1]
```diff
@@ -1065,8 +1065,12 @@ async def _create_jobs(userdata: dict, job_specs: dict, batch_id: int, update_id
             spec['env'].append({'name': 'AZURE_APPLICATION_CREDENTIALS', 'value': '/gsa-key/key.json'})
 
         cloudfuse = spec.get('gcsfuse') or spec.get('cloudfuse')
-        if not is_developer and user not in ('ci', 'test', 'test-dev') and cloudfuse is not None and len(cloudfuse) > 0:
-            raise web.HTTPBadRequest(reason='cloudfuse requests are temporarily not supported.')
+        if cloudfuse:
+            for config in cloudfuse:
+                if not config['read_only']:
+                    raise web.HTTPBadRequest(reason='Only read-only cloudfuse requests are supported.')
```
Comment (danking) [inline batch/batch/front_end/front_end.py:1071]: Include the bucket name and mount point to aid debugging.

File: batch/batch/front_end/front_end.py [hunk 2]
```diff
@@ -1065,8 +1065,12 @@ async def _create_jobs(userdata: dict, job_specs: dict, batch_id: int, update_id
...
+                if config['mount_path'] == '/io':
+                    raise web.HTTPBadRequest(reason='Cloudfuse requests with mount_path=/io are not supported.')
```
Comment (danking) [inline batch/batch/front_end/front_end.py:1073]: likewise include bucket name to aid debugging. Maybe just stick the whole `config` in there as a JSON string?

File: hail/python/test/hailtop/batch/test_batch.py [hunk 3]
```diff
@@ -686,23 +688,37 @@ def test_verbose(self):
...
-        tail = b.new_job()
-        tail.command(f'cat {path}/cloudfuse_test_1')
-        tail.cloudfuse(self.bucket, f'/{self.bucket}', read_only=True)
-        tail.depends_on(head)
+        try:
+            b.run()
+        except aiohttp.ClientResponseError as e:
+            assert e.message == 'Only read-only cloudfuse requests are supported.', e.message
+        else:
+            assert False
```
Comment (danking) [inline hail/python/test/hailtop/batch/test_batch.py:705]: I recently learned about `with pytest.raises(...):`. This is fine, but thought I'd share!

**Review by danking — CHANGES_REQUESTED** [review]

**Review by danking — APPROVED** [review]
LGTM!

**Comment by daniel-goldstein** [comment]
@jigold Lint failed because the `is_developer` variable is no longer used

**Pull request merged**